tests/rptest/services/redpanda_installer.py

```diff
@@ -454,8 +482,72 @@ def latest_for_line(self, release_line: tuple[int, int]):
 
         assert False, f"no downloadable versions in {versions_in_line[0:2]} for {release_line=}"
 
-    def install(self, nodes, version: typing.Union[str, tuple[int, int],
-                                                   tuple[int, int, int]]):
+    def release_range(
+        self,
+        *,
+        earliest: typing.Optional[RedpandaVersionLine
+                                  | RedpandaVersionTriple] = None,
+        latest: RedpandaVersion = REDPANDA_INSTALLER_HEAD_TAG,
+        do_not_skip: list[RedpandaVersionTriple] = []
+    ) -> list[RedpandaVersionTriple]:
+        """
+        returns a list of version to implement an upgrade path. for each line, it will return the latest for the line,
+        from earliest up to latest line.
+        versions in the do_not_skip list will be integrated in the upgrade path
+        e.g release_range(earliest=(21, 2), latest=RedpandaInstaller.HEAD, do_not_skip=[(22, 2, 7)]) ->
+            [(21, 2, 8), (21, 3, 9), (22, 1, 7), (22, 2, 7), (22, 2, 8), (22, 3, 9), (23, 1, 0)]
+
+        :param earliest: first version or line to install DEFAULT the oldest line
+        :param latest: last version or line or HEAD to install DEFAULT RedpandaInstallerHEAD
+        :param do_not_skip: versions to integrate in the list DEFAULT empty
+        :return: a list of version in the range [earliest, latest],
+            each being a) the latest from its line or b) one from do_not_skip.
+        """
+
+        self.start()
+
+        # default to oldest
+        if not earliest:
+            earliest = self.released_versions[-1][0:2]
+
+        # convert to specific version
+        if len(earliest) == 2:
+            earliest, _ = self.latest_for_line(earliest)
+
+        # default to _head_version
+        if latest == RedpandaInstaller.HEAD:
+            latest = self._head_version
+
+        assert (isinstance(latest, tuple))
+
+        # convert to specific version
+        if len(latest) == 2:
+            latest, _ = self.latest_for_line(latest)
+
+        # setup a copy (we are going to modify it)
+        rel_range = copy.deepcopy(self.released_versions)
+
+        # glue head in front, as the most recent version
+        rel_range.insert(0, self._head_version)
+
+        # clamp to [latest, earliest]
+        rel_range = rel_range[rel_range.index(latest
```

Comment:
added, but I'm thinking of removing this function altogether in favor of RedpandaTest::load_version_range https://github.com/redpanda-data/redpanda/blob/dev/tests/rptest/tests/redpanda_test.py#L120
since this one is a bit overcomplicated, considering that, I only use the param earliest